src/Pos/Block/Arbitrary.hs

```diff
@@ -220,8 +219,7 @@ newtype BlockHeaderList ssc = BHL
     } deriving (Eq)
 
 instance T.BiSsc ssc => Show (BlockHeaderList ssc) where
-    show =
-        concat . intersperse "\n" . map (formatToString build) . uncurry zip . getHeaderList
+    show = intercalate "\n" . map (formatToString build) . uncurry zip . getHeaderList
```

Comment:
Well, AFAIK, our philosophy is that you should use `Text` instead of `String` as much as possible. My version does everything with `Text` and converts result to `String` only in the end. I expect it to be more efficient, for example (though performance shouldn't be a concern here).